### PR TITLE
Make source patches configurable

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: In one of our projects we want to provide patched static
tezos-binaries.

Solution: Make patches configurable.

## Related issue(s)

None

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -365,7 +365,7 @@ let
 
         minimumOCamlVersion = "4.07";
 
-        patches = [ branchInfo.patchFile ];
+        patches = [ branchInfo.patches ];
```
Comment (gromakovsky) [inline nix/default.nix]: AFAIU it was a list of patches (with one item), but now it is a list of lists of patches, but I barely know `nix`. Can you clarify where it is equivalent? If it was Haskell I would say it should be `patches = branchInfo.patches`

Comment (rvem) [reply nix/default.nix]: :think: this looks like a bug, but nix somehow figured it out

Comment (gromakovsky) [reply nix/default.nix]: So I'd like to know whether nix accidentally figured it out or it is reliable and specified behavior.

Comment (rvem) [reply nix/default.nix]: I assume, that it performs list flattening somewhere, but I haven't found any proofs yet

Comment (gromakovsky) [reply nix/default.nix]: Anyway, don't you think that `patches = branchInfo.patches` would be less confusing?

Comment (rvem) [reply nix/default.nix]: yeah, I agree

**Pull request merged**